File: life/lib/models/usuario.dart
class Usuario {
  final String? id;
  final String nombre;
  final String apellido;
  final String telefono;
  final String correo;
  final String age;
  final String password;

  Usuario({
    this.id,
    required this.nombre,
    required this.apellido,
    required this.telefono,
    required this.correo,
    required this.age,
    required this.password,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': nombre,
      'surname': apellido,
      'telephone': telefono,
      'email': correo,
      'age': age,
      'password': password,
    };
  }

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['_id'],
      nombre: json['name'],
      apellido: json['surname'],
      telefono: json['telephone'],
      correo: json['email'],
      age: json['age'],
      password: '',
    );
  }
}

File: backend/src/models/Usuario.ts
import mongoose from 'mongoose';

export const usuarioSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        unique: false,
    },
    surname: {
        type: String,
        required: true,
        unique: false
    },
    telephone: {
        type: String,
        required: true,
        unique: true
    },
    email: {
        type: String,
        required: true,
        unique: true,
        match: [/^\S+@\S+\.\S+$/, 'Email inválido'],
        trim: true
    },
    age: {
        type: Number,
        required: true,
        unique: false
    },
    password: {
        type: String,
        required: true,
        unique: false
    }
}, {timestamps: true

});

export default mongoose.model('Usuario', usuarioSchema);
